### PR TITLE
Set gas limit to 21k for recipients with no code

### LINKED ISSUE #2907: Enhancement: Use 21k reccomended gasLimit when tx recipient has no code
check for code via `eth_getCode` before running `eth_estimateGas`

if no code, return 21000 as estimated `gasLimit`

### PR DESCRIPTION
**Pull request by danfinlay**
Fixes #2907 

Also fixes strange behavior where before, buffer was added to transactions and VM was used even when web3 user specified a `gas` value.

Allows dapp devs to hard-code intelligent gas limits to bypass gas estimating time during any future `estimateGas` slowdowns.

### CODE REVIEW

File: app/scripts/lib/tx-gas-utils.js [hunk 1]
```diff
@@ -37,14 +38,30 @@ module.exports = class txProvideUtil {
...
+      return txParams.gas
     }
+
+    // if recipient has no code, gas is 21k max:
+    const recipient = txParams.to
+    const hasRecipient = Boolean(recipient)
+    const code = await this.query.getCode(recipient)
+    if (hasRecipient && (!code || code === '0x')) {
+      txParams.gas = SIMPLE_GAS_COST
+      txMeta.gasLimitSpecified = true // Prevents buffer addition
```
Comment (kumavis) [inline app/scripts/lib/tx-gas-utils.js]: `gasLimitSpecified` is used to suggest that the dapp suggested the `gasLimit`, lets not overload that meaning

Comment (kumavis) [reply app/scripts/lib/tx-gas-utils.js]: seems that bc we return here we dont add the buffer addition, and not because of setting the flag

Comment (danfinlay) [reply app/scripts/lib/tx-gas-utils.js]: No, the buffer is added even if you simply return here.

Comment (danfinlay) [reply app/scripts/lib/tx-gas-utils.js]: > lets not overload that meaning

I could just add another boolean flag that means the same thing?

Comment (danfinlay) [reply app/scripts/lib/tx-gas-utils.js]: Whatever is returned from this method is passed to `setTxGas`, which adds a buffer unless the flag is set.

Comment (kumavis) [reply app/scripts/lib/tx-gas-utils.js]: ah i see. logic flow is sort of strange here, could use some cleanup. e.g. we set to `19/20 * block.gasLimit` then override that with an `estimateGas`


```js
if (txMeta.gasLimitSpecified || txParams.gas === SIMPLE_GAS_COST) {
  //  ...
}
```

**Comment by kumavis** [comment]
Yeah looks good for now -- I do think this whole section is due for a refresh

**Review by kumavis — APPROVED** [review]

**Pull request merged**